src/PostgREST/Cors.hs

```diff
@@ -44,3 +44,7 @@ corsPolicy corsAllowedOrigins req = case lookup "origin" headers of
        -- Impossible case, Middleware.Cors will not evaluate this when
        -- the Access-Control-Request-Headers header is not set.
       Nothing   -> []
+    -- The library makes "OPTIONS" requests fail when the "Origin" header is present
```

Comment:
Hm, which library?

Reply:
Ah, I failed to mention the `wai-cors` library. 

I added more info here: https://github.com/PostgREST/postgrest/issues/3027#issuecomment-1815405884. The solution is as I mentioned there or patch the library to remove that validation.

Reply:
Just confirmed that there's no way to bypass the `OPTIONS` requests without re-implementing the CORS functionality for that method only. So, I opened an issue upstream https://github.com/larskuhtz/wai-cors/issues/37 to see what they think about it.